construction_impl.go

```diff
@@ -15,50 +15,112 @@
 ********************************************************************************/
 package rosetta_filecoin_lib
 
-import "github.com/filecoin-project/go-address"
-import "github.com/filecoin-project/lotus/lib/sigs"
-import "github.com/filecoin-project/specs-actors/actors/crypto"
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/lotus/chain/types"
+	c "github.com/filecoin-project/go-crypto"
+	"github.com/filecoin-project/specs-actors/actors/crypto"
+	"github.com/minio/blake2b-simd"
+	cbg "github.com/whyrusleeping/cbor-gen"
+	"encoding/json"
+)
 
 type RosettaConstructionFilecoin struct {
 	Mainnet bool
 }
 
+func signSecp256k1(msg []byte, pk []byte) ([]byte, error) {
+	b2sum := blake2b.Sum256(msg)
+	sig, err := c.Sign(pk, b2sum[:])
+	if err != nil {
+		return nil, err
+	}
+
+	return sig, nil
+}
+
+// REVIEW: Doesn't actually verify the signature...
+// https://github.com/filecoin-project/lotus/blob/master/lib/sigs/secp/init.go#L38-L55
+func verifySecp256k1(sig []byte, a address.Address, msg []byte) error {
+	b2sum := blake2b.Sum256(msg)
+	pubk, err := c.EcRecover(b2sum[:], sig)
+	if err != nil {
+		return err
+	}
+
+	maybeaddr, err := address.NewSecp256k1Address(pubk)
+	if err != nil {
+		return err
+	}
+
+	if a != maybeaddr {
+		return fmt.Errorf("signature did not match")
+	}
+
+	if c.Verify(pubk, b2sum[:], sig) {
```

Comment:
It didn't feel rigt to not verify the signature so I have added this line to their function.